src/algorithms/pid_lut/PIDLookup.cc

```diff
@@ -126,7 +126,7 @@ void PIDLookup::process(const Input& input, const Output& output) const {
         identified_pdg = 321; // kaon
         recopart.setParticleIDUsed((*partids_out)[partids_out->size() - 2]);
       } else if (random_unit_interval < (entry->prob_electron + entry->prob_pion +
-                                         entry->prob_kaon + entry->prob_electron)) {
+                                         entry->prob_kaon + entry->prob_proton)) {
```

Comment:
The underlying problem here is probably that we encourage copy & paste. We should not have `entry` be a struct with fields per particle type. Can this be treated as an array with a more programmatic approach that is sure to not lead to copy/paste errors?

Reply:
Probably. Even if that's the case, there is no point in future-proofing PID lookup codebase.